main/src/ca/uwaterloo/flix/language/phase/jvm/GenFunAndClosureClasses.scala

```diff
@@ -146,40 +146,66 @@ object GenFunAndClosureClasses {
     */
   private def genCode(className: JvmName, kind: FunctionKind, defn: Def)(implicit root: Root, flix: Flix): Array[Byte] = {
     val visitor = AsmOps.mkClassWriter()
+    kind match {
+      case Function if Purity.isControlPure(defn.expr.purity) =>
+        // Header
+        val functionInterface = JvmOps.getErasedFunctionInterfaceType(defn.arrowType).jvmName
```

Comment:
would it make sense to extract these three branches into functions? then this function is only responsible for choosing pureDef/controlDef/closure and those functions are responsible for implementing that class

Reply:
I guess I'm essentially saying that this reused genCode with a Kind parameter is kind of useless now because the versions are so different